openwrt-openwisp-monitoring/files/monitoring.agent

```diff
@@ -26,76 +29,179 @@ if [ "$VERSION" -eq "1" ]; then
   exit 0
 fi
 
-VERIFY_SSL=${VERIFY_SSL:-0}
 INTERVAL=${INTERVAL:-300}
-MONITORED_INTERFACES=${MONITORED_INTERFACES:-*}
 verbose_mode=${verbose_mode:-0}
+TMP_DIR="/tmp/openwisp/monitoring"
 
-if [ -z "$BASE_URL" ]; then
-  logger -s "missing required --url option" \
-         -t openwisp_monitoring \
-         -p daemon.err
-  exit 1
-fi
+echoerr() { echo "$@" 1>&2; }
 
-if [ -z "$UUID" ]; then
-  logger -s "missing required --uuid option" \
-          -t openwisp_monitoring \
-          -p daemon.err
-  exit 1
-fi
+check_available_memory(){
+  total=$(ubus call system info | jsonfilter -e '@.memory.total')
+  available=$(ubus call system info | jsonfilter -e '@.memory.available')
+  required=$(echo - |awk -v percent="$REQUIRED_PERCENT" -v total="$total" '{printf("%.f",percent*total)}')
 
-if [ -z "$KEY" ]; then
-  logger -s "missing required --key option" \
-         -t openwisp_monitoring \
-         -p daemon.err
-  exit 1
-fi
-
-# Remove double quotes from interfaces
-MONITORED_INTERFACES=$(echo "$MONITORED_INTERFACES" | tr -d '"')
-
-URL="$BASE_URL/api/v1/monitoring/device/$UUID/?key=$KEY"
-
-CURL_COMMAND="curl -s -w "%{http_code}""
-[ "$VERIFY_SSL" = "0" ] && CURL_COMMAND="$CURL_COMMAND -k"
-[ "$verbose_mode" = "1" ] && CURL_COMMAND="$CURL_COMMAND -v"
+  if [ "$available" -ge "$required" ]; then
+    echo "1"
+  else
+    [ "$verbose_mode" -eq "1" ] && logger -s "Not enough memory available" \
+                                          -p daemon.err
+    echo "0"
```

Comment:
I think this should be changed too.
0 should be returned if case of success, else we should return 1

Reply:
Changed